panic: non-positive interval for NewTicker

**Describe the bug**
Since a few weeks (without updating lazygit itself), I get the following error when starting lazygit from the shell:

panic: non-positive interval for NewTicker

goroutine 11 [running]:
time.NewTicker(0x0, 0x0)
	/opt/homebrew/Cellar/go/1.16.4/libexec/src/time/tick.go:24 +0x144
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).goEvery.func1()
	/private/tmp/lazygit-20210604-23703-e7fn5f/lazygit-0.28.2/pkg/gui/gui.go:697 +0x4c
github.com/jesseduffield/lazygit/pkg/utils.Safe.func1(0x0, 0x0)
	/private/tmp/lazygit-20210604-23703-e7fn5f/lazygit-0.28.2/pkg/utils/utils.go:107 +0x28
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0x140002727b8, 0x0, 0x0)
	/private/tmp/lazygit-20210604-23703-e7fn5f/lazygit-0.28.2/pkg/utils/utils.go:118 +0x5c
github.com/jesseduffield/lazygit/pkg/utils.Safe(0x14000079050)
	/private/tmp/lazygit-20210604-23703-e7fn5f/lazygit-0.28.2/pkg/utils/utils.go:107 +0x40
created by github.com/jesseduffield/lazygit/pkg/gui.(*Gui).goEvery
	/private/tmp/lazygit-20210604-23703-e7fn5f/lazygit-0.28.2/pkg/gui/gui.go:696 +0x88

**Desktop (please complete the following information):**
 - OS: OSX 11.5.1, M1 Macbook Pro, Homebrew
 - Lazygit Version 0.28.2 via homebrew



Thank You for your help!

> @henkworks Auto fetching can be disabled by setting the `autoFetch` option in the config file to `false`.
> 
> ```
> git:                                                                                                
>   autoFetch: false
> ```

Thank you very much, that helps with my main concern!

The issue this bug is about is still present with that setting though, as long as the following is in the config, it seems:
```
refresher:
  refreshInterval: 0
```
produces:
```
± ~/go/bin/lazygit
panic: non-positive interval for NewTicker

goroutine 18 [running]:
time.NewTicker(0x0, 0x4db02f)
        /usr/lib/go-1.15/src/time/tick.go:23 +0x151
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).goEvery.func1()
        /home/henk/Data/public/software/lazygit/pkg/gui/gui.go:692 +0x67
github.com/jesseduffield/lazygit/pkg/utils.Safe.func1(0xc0001e6840, 0xc0000437b0)
        /home/henk/Data/public/software/lazygit/pkg/utils/utils.go:95 +0x24
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0xc0000437b8, 0x0, 0x0)
        /home/henk/Data/public/software/lazygit/pkg/utils/utils.go:106 +0x62
github.com/jesseduffield/lazygit/pkg/utils.Safe(0xc000090600)
        /home/henk/Data/public/software/lazygit/pkg/utils/utils.go:95 +0x49
created by github.com/jesseduffield/lazygit/pkg/gui.(*Gui).goEvery
        /home/henk/Data/public/software/lazygit/pkg/gui/gui.go:691 +0x8a
[2]    1585044 exit 2     ~/go/bin/lazygit
```

--- a/docs/Config.md
+++ b/docs/Config.md
@@ -86,7 +86,7 @@ os:
   openCommand: ''
 refresher:
   refreshInterval: 10 # File/submodule refresh interval in seconds. Auto-refresh can be disabled via option 'git.autoRefresh'.
-  fetchInterval: 60 # re-fetch interval in seconds
+  fetchInterval: 60 # Re-fetch interval in seconds. Auto-fetch can be disabled via option 'git.autoFetch'.
 update:
   method: prompt # can be: prompt | background | never
   days: 14 # how often an update is checked for

--- a/docs/Config.md
+++ b/docs/Config.md
@@ -73,6 +73,7 @@ git:
     showGraph: 'when-maximised'
   skipHookPrefix: WIP
   autoFetch: true
+  autoRefresh: true
   branchLogCmd: 'git log --graph --color=always --abbrev-commit --decorate --date=relative --pretty=medium {{branchName}} --'
   allBranchesLogCmd: 'git log --graph --all --color=always --abbrev-commit --decorate --date=relative  --pretty=medium'
   overrideGpg: false # prevents lazygit from spawning a separate process when using GPG
@@ -84,7 +85,7 @@ os:
   editCommandTemplate: '{{editor}} {{filename}}'
   openCommand: ''
 refresher:
-  refreshInterval: 10 # file/submodule refresh interval in seconds
+  refreshInterval: 10 # File/submodule refresh interval in seconds. Auto-refresh can be disabled via option 'git.autoRefresh'.
   fetchInterval: 60 # re-fetch interval in seconds
 update:
   method: prompt # can be: prompt | background | never

--- a/pkg/config/user_config.go
+++ b/pkg/config/user_config.go
@@ -68,6 +68,7 @@ type GitConfig struct {
 	Merging             MergingConfig                 `yaml:"merging"`
 	SkipHookPrefix      string                        `yaml:"skipHookPrefix"`
 	AutoFetch           bool                          `yaml:"autoFetch"`
+	AutoRefresh         bool                          `yaml:"autoRefresh"`
 	BranchLogCmd        string                        `yaml:"branchLogCmd"`
 	AllBranchesLogCmd   string                        `yaml:"allBranchesLogCmd"`
 	OverrideGpg         bool                          `yaml:"overrideGpg"`
@@ -373,6 +374,7 @@ func GetDefaultConfig() *UserConfig {
 			},
 			SkipHookPrefix:      "WIP",
 			AutoFetch:           true,
+			AutoRefresh:         true,
 			BranchLogCmd:        "git log --graph --color=always --abbrev-commit --decorate --date=relative --pretty=medium {{branchName}} --",
 			AllBranchesLogCmd:   "git log --graph --all --color=always --abbrev-commit --decorate --date=relative  --pretty=medium",
 			DisableForcePushing: false,

--- a/pkg/gui/gui.go
+++ b/pkg/gui/gui.go
@@ -587,7 +587,16 @@ func (gui *Gui) Run(filterPath string) error {
 		go utils.Safe(gui.startBackgroundFetch)
 	}
 
-	gui.goEvery(time.Second*time.Duration(userConfig.Refresher.RefreshInterval), gui.stopChan, gui.refreshFilesAndSubmodules)
+	if userConfig.Git.AutoRefresh {
+		refreshInterval := userConfig.Refresher.RefreshInterval
+		if refreshInterval > 0 {
+			gui.goEvery(time.Second*time.Duration(refreshInterval), gui.stopChan, gui.refreshFilesAndSubmodules)
+		} else {
+			gui.c.Log.Errorf(
+				"Value of config option 'refresher.refreshInterval' (%d) is invalid, disabling auto-refresh",
+				refreshInterval)
+		}
+	}
 
 	gui.c.Log.Info("starting main loop")
 

--- a/pkg/gui/gui.go
+++ b/pkg/gui/gui.go
@@ -583,8 +583,16 @@ func (gui *Gui) Run(filterPath string) error {
 	}
 
 	gui.waitForIntro.Add(1)
-	if gui.c.UserConfig.Git.AutoFetch {
-		go utils.Safe(gui.startBackgroundFetch)
+
+	if userConfig.Git.AutoFetch {
+		fetchInterval := userConfig.Refresher.FetchInterval
+		if fetchInterval > 0 {
+			go utils.Safe(gui.startBackgroundFetch)
+		} else {
+			gui.c.Log.Errorf(
+				"Value of config option 'refresher.fetchInterval' (%d) is invalid, disabling auto-fetch",
+				fetchInterval)
+		}
 	}
 
 	if userConfig.Git.AutoRefresh {

